### PR TITLE
github-auth

### LINKED ISSUE #1057: feature: make user link their github here
Screenshot 2024-08-27 at 5 07 07 PM

Should be a button that says `link your github` that takes user through auth flow to link it.

That way, whenever a bounty is given it can automatically reach the specific users solana address

### PR DESCRIPTION
**Pull request by Ashutoshpadhi629**
### PR Fixes:
- updated github Oauth 
- fixed Calendar UI

Resolves #1057 #1056 

https://github.com/user-attachments/assets/f9ea8677-64f2-47f1-9af6-74d06011557e

https://github.com/user-attachments/assets/921b58d9-1f98-4ceb-8796-a735d97ec240

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I assure there is no similar/duplicate pull request regarding same issue

### CODE REVIEW

**Comment by devsargam** [comment]
Few questions, for example if a user logged in with github then how does it log into website to access the course dashboard?

**Comment by devsargam** [comment]
Also, @hkirat is this feature to be included in cms? or can we add this to some other place where we just aggregate the data to notion file

File: prisma/schema.prisma [hunk 1]
```diff
@@ -21,7 +21,7 @@ model Course {
   content         CourseContent[]
```
Comment (devsargam) [inline prisma/schema.prisma:21]: This file has a lot of noise.

**Comment by Ashutoshpadhi629** [comment]
> Few questions, for example if a user logged in with github then how does it log into website to access the course dashboard?

no , the next Auth config is not changed , this github Oauth flow is completely different from that.

File: src/actions/payoutMethods/index.ts [hunk 2]
```diff
@@ -233,3 +234,36 @@ export const deleteSolanaAddress = createSafeAction(
...
 );
+export const githubRedirectAddress = () => {
+  const url = `https://github.com/login/oauth/authorize?client_id=${process.env.GITHUB_ID}`;
+  redirect(url);
+};
+export const getGithubAddress = async () => {
+  const session = await getServerSession(authOptions);
+
+  if (!session || !session.user.id) {
+    throw new Error('Unauthorized');
```
Comment (devsargam) [inline src/actions/payoutMethods/index.ts:245]: Don't directly throw an error. Just send 401 status code with "Unauthenticated" message

File: src/actions/payoutMethods/index.ts [hunk 3]
```diff
@@ -233,3 +234,36 @@ export const deleteSolanaAddress = createSafeAction(
...
+      id: session.user.id,
+    },
+  });
+  return user?.github;
+};
+export const deleteGithubAddress = async () => {
+  const session = await getServerSession(authOptions);
+
+  if (!session || !session.user.id) {
+    throw new Error('Unauthorized');
```
Comment (devsargam) [inline src/actions/payoutMethods/index.ts:259]: Save as above

**Review by devsargam — CHANGES_REQUESTED** [review]

**Comment by devsargam** [comment]
@Ashutoshpadhi629 here are a few follow ups. Mention me again when u resolve all 3 for re-review

**Comment by devsargam** [comment]
Closing in favour of https://github.com/code100x/cms/pull/1066

**Pull request closed without merging**